src/tools/dotnet.lua

```diff
@@ -50,7 +49,11 @@
 			info.action = "Resource"
 		elseif ext == ".xaml" then
 			if fcfg.buildaction == "Application" or path.getbasename(fname) == "App" then
-				info.action = "ApplicationDefinition"
+				if fcfg.project.kind ~= premake.SHAREDLIB then
```

Comment:
Personal opinion: This would make more sense inversed since there's an else and this is a `~=`.